### PR TITLE
ROC-2473: added api logic and unit tests. Nofitication sending to cla…

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-2473

### Change description ###
Notification sent when offer made

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/OfferMadeEvent.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import uk.gov.hmcts.cmc.claimstore.models.Claim;
+
+import java.util.Objects;
+
+public class OfferMadeEvent {
+
+    private final Claim claim;
+
+
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/events/OfferMadeEvent.java]: extra line here

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+@Component
+public class OfferResponseDeadlineCalculator {
+
+    private final WorkingDayIndicator workingDayIndicator;
+    private final int offerMadeTimeForResponseInDays;
+    private final int endOfBusinessDayHour;
+
+    public OfferResponseDeadlineCalculator(
+        WorkingDayIndicator workingDayIndicator,
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java]: two values final but not this one?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        @Value("${dateCalculations.offerMadeTimeForResponseInDays}") final int offerMadeTimeForResponseInDays,
+        @Value("${dateCalculations.endOfBusinessDayHour}") final int endOfBusinessDayHour) {
+
+        this.workingDayIndicator = workingDayIndicator;
+        this.offerMadeTimeForResponseInDays = offerMadeTimeForResponseInDays;
+        this.endOfBusinessDayHour = endOfBusinessDayHour;
+    }
+
+    public LocalDate calculateOfferResponseDeadline(LocalDateTime offerSentAt) {
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java]: there's a mix of final parameters and non final in this class?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/events/OfferMadeEvent.java (hunk 1 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/events/OfferMadeEvent.java]: removed

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java (hunk 2 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java]: `final` added

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java (hunk 3 above)
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java]: all are `final` now

**Pull request merged**